x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_sync_rules_table_saved_state.ts

```diff
@@ -50,6 +50,11 @@ export function useSyncRulesTableSavedState(): void {
       storageStateToSave.tags = state.filterOptions.tags;
     }
 
+    if (state.filterOptions.enabled !== undefined) {
+      urlStateToSave.enabled = state.filterOptions.enabled;
+      storageStateToSave.enabled = state.filterOptions.enabled;
+    }
+
```

Comment:
Commenting here since this is where `sessionStorage.set()` is called below to set the local storage values.

In testing, I had cleared my localstorage via devtools, and now I'm not seeing it be set again? I did a clean deployment and am still seeing this behavior. I'm seeing the urlParams get set, but not localstorage... 🤔 


When debugging, for some reason `RULES_TABLE_STATE_STORAGE_KEY` is showing as `undefined` (even though it is definitely in the `constants` file).

<img width="447" alt="image" src="https://user-images.githubusercontent.com/2946766/216462343-4c51130b-74f8-43a2-bc7c-6643ad3f5ece.png">

---

Also seeing this Typescript error, but when debugging it seems to go into that function and work fine:

<img width="459" alt="image" src="https://user-images.githubusercontent.com/2946766/216462995-1c9364bb-9291-4d7f-82d4-536d45ff277c.png">

Reply:
@spong thank you for the review!

Could you describe a reproduction scenario when you don't see the state restored in the session storage?Basically something should happen like a page reload or state update. I've tested in Chrome and FF and it works well though I have issues with displaying all the session and local storage values. It worked well after reinstalling Chrome but I have the same problem again.

Reply:
>Could you describe a reproduction scenario when you don't see the state restored in the session storage?

> Basically something should happen to update the saved state in the session storage, for example a page reload or state update. I've tested in Chrome and FF and it works well though I have issues with displaying all the session and local storage values in newest Chrome. It worked well after reinstalling Chrome but I have the same problem again. Maybe you experience the same.

Interesting... 🤔 As for repro steps, the only thing I did here was manually delete the `securitySolution.rulesTable` key under `Application -> Local Storage` in Dev Tools (at least I recall it being Local Storage, not Session). After poking around a bit more though, it seems it's being added back to the `Session Storage` section in dev tools instead of `Local Storage` where the other `securitySolution` keys are:

##### Local Storage
 <img width="1324" alt="image" src="https://user-images.githubusercontent.com/2946766/217064276-ce6bc9c8-b319-4151-8a93-9a1af2b87fd3.png">

##### Sessions Storage
<img width="1331" alt="image" src="https://user-images.githubusercontent.com/2946766/217064580-26969048-f5e6-4c23-ab16-d37f4baa641f.png">

I guess this explains why the app is working as expected, and I didn't notice that it was storing to session vs local storage. I was pretty sure I first saw the key in Local Storage, but I could be mistaken here. Either way, sounds like everything is 👍 here and working as expected.

---

> Regarding unavailability of RULES_TABLE_STATE_STORAGE_KEY value for debugging. I experience the same but it looks related to the dev tool or it's transpiled the way when constants are inlined.
>
> Regarding TS errors it seems like a local issue. I don't see anything like that. If you just switched a branch and IDE hasn't updated its cache or installed another TS version locally it can be an issue.

Yeah that was my thought as well (as I often see type issues the end up getting resolved when clicking through to the source). Though with this now merged on main and after a fresh bootstrap, I'm still seeing the same type error. Modifying line `34` to be `const urlStateToSave: RisonValue = {};` instead of `RulesTableUrlSavedState` resolves the error, or of course `as` casting `urlStateToSave as RisonValue` inline on line `93` as well. So seems `RulesTableUrlSavedState` should extend from `RisonValue` or we should update our types in `useReplaceUrlParams()` then?